website/docs/feature-flag-tutorials/use-cases/trunk-based-development.md

```diff
@@ -0,0 +1,222 @@
+
+### Merge Every Commit
+
+The most frequent approach to trunk-based development is merging every commit directly into the main branch. This strategy maximizes the rate of continuous integration and delivery, as changes are integrated into the trunk as soon as they are ready. Some advantages are:
+
+-   Rapid feedback loops and quicker issue resolution
+-   Reduced risk of large, complex merges
+-   Encourages small, frequent commits
+
+Some challenges with merging every commit are:
+
+-   Potential for more frequent build breaks and test failures
+-   Increased need for effective merging strategies
+-   Requirement for robust automated testing and CI/CD pipelines
+
+### Merge at Set Intervals
+
+Another common strategy is to merge changes into the trunk at regular, predefined intervals, such as daily or before the end of each work day. This approach offers a balance between frequent integration and managing work-in-progress.
+
+**Advantages**:
+
+-   Improved control over the integration process
+-   Reduced risk of breaking the build with frequent merges
+-   Easier to manage and identify integration issues
+
+**Challenges**:
+
+-   Longer feedback loops compared to merging every commit
+-   Potential for larger, more complex merges
+-   Requires discipline to adhere to the defined merge cadence
+
+### Benefits of Short-Lived Feature Branches
+
+In trunk-based development, it is possible to use short-lived feature branches, which enable teams to implement code review processes without the burden of maintaining long-running branches, which can become increasingly difficult to manage over time. The frequent integration back to trunk significantly reduces the likelihood and complexity of merge conflicts, as changes are merged before they can diverge too far from the main codebase. This approach maintains the core benefits of trunk-based development - such as rapid iteration and continuous integration - while still allowing developers to work in isolation when needed. This isolation can be particularly valuable for making experimental changes or implementing complex features that require validation before being merged into the trunk.
+
+**Challenges to Address**
+
+The most significant challenge is maintaining the discipline required to keep branches truly short-lived. Teams must develop the skill of breaking down work into small, mergeable units. Additionally, this approach demands a robust testing and CI/CD pipeline to ensure that frequent merges don't compromise code quality or stability. Teams must invest in automated testing and deployment infrastructure to support rapid integration cycles.
+
+Regardless of the specific strategy, Unleash feature flag management capabilities are designed to support a wide range of trunk-based development styles, empowering you to implement the best strategy for your teams.
+
+### Microservices vs. Monoliths
```

Comment:
This is bit of a hairy topic. I think all the points you've listed here can be challenged because they depend on a lot of things, like how well the code itself is structured and how source control is done. Are you mentioning this for completeness?

Not against this, just not sure where we're going with the thought

Reply:
This was suggested to be mentioned so I asked a customer what their thoughts on this topic was in relation to feature flags. Microservices vs. monoliths is not directly linked to trunk-based with feature flags, but I think this is a snippet that could be expanded to its own post and maybe linked from this doc to the future one. Definitely a provocative conversation-starter topic.